Freetz toolchain changes gcc's default from --no-as-needed to --as-needed.

uClibc Makefile however relies on --no-as-needed being the default (to be more precise the linking order is not correct everywhere).
Instead of fixing the linking order (requires more effort) add an explicit --no-as-needed to fix the problem.

--- Rules.mak
+++ Rules.mak
@@ -626,7 +626,7 @@
 $(eval $(call check-ld-var,--warn-once))
 $(eval $(call check-ld-var,--sort-common))
 $(eval $(call check-ld-var,--discard-all))
-LDFLAGS_NOSTRIP:=$(LDFLAG-fuse-ld) $(CPU_LDFLAGS-y) -shared \
+LDFLAGS_NOSTRIP:=$(CC_FLAG_NO_ASNEEDED) $(LDFLAG-fuse-ld) $(CPU_LDFLAGS-y) -shared \
 	-Wl,--warn-common $(CFLAG_-Wl--warn-once) -Wl,-z,combreloc
 # binutils-2.16.1 warns about ignored sections, 2.16.91.0.3 and newer are ok
 #$(eval $(call check-ld-var,--gc-sections))
